### PR TITLE
CAMS-269 Initial setup of end-to-end integration test with Playwright

### PR DESCRIPTION
**Pull request by magile32**
# Purpose

Experimentation with e2e testing solution and provide initial implementation running in CI pipeline.

# Major Changes

- Setup integration testing with Playwright
- Update CICD workflow to execute integration test against deployed webapp in Flexion environment

# Testing/Validation

- Verify passing integration test locally and in CICD workflow

# Notes

### CODE REVIEW

File: test/e2e/playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    //   name: 'firefox',
+    //   use: { ...devices['Desktop Firefox'] },
+    // },
+    //
+    // {
+    //   name: 'webkit',
+    //   use: { ...devices['Desktop Safari'] },
+    // },
+
+    /* Test against mobile viewports. */
```
Comment (amorrow-flexion) [inline test/e2e/playwright.config.ts:53]: should we be keeping these commented out portions here?

File: test/e2e/playwright/transfer-orders.spec.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  await page.getByTestId(`new-case-input-${firstOrderId}`).fill('11-11111');
+  await expect(page.getByTestId('alert-container-validation-not-found')).toBeVisible();
+
+  const caseNumber = '18-61881';
+  const summaryRequestPromise = page.waitForEvent('requestfinished', {
+    predicate: (e) => e.url().includes(`/api/cases/081-${caseNumber}/summary`),
+  });
+  await page.getByTestId(`new-case-input-${firstOrderId}`).fill(caseNumber);
+  await summaryRequestPromise;
+  // await page.on('requestfinished', (data) => data.response() !== null);
```
Comment (amorrow-flexion) [inline test/e2e/playwright/transfer-orders.spec.ts:116]: remove this?

File: test/e2e/playwright/transfer-orders.spec.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  // console.log('debugging', firstOrder.docketSuggestedCaseNumber);
+  // await expect(page.getByTestId(`new-case-input-${firstOrderId}`)).toHaveValue(
+  //   firstOrder.docketSuggestedCaseNumber,
+  // );
+  await expect(page.getByLabel(`Select new court`).locator('visible=true')).toHaveValue('');
+
+  const page1Promise = page.waitForEvent('popup');
+  await page.getByTestId(`approved-transfer-original-case-link-${firstOrder.caseId}-link`).click();
+  const page1 = await page1Promise;
+  // const page1DetailPromise = page1.waitForEvent('requestfinished', {
```
Comment (amorrow-flexion) [inline test/e2e/playwright/transfer-orders.spec.ts:159]: Is there a reason for keeping this around?

File: test/e2e/.github/workflows/playwright.yml
[the remark this answers is not in the document]
Comment (magile32) [reply test/e2e/.github/workflows/playwright.yml:1]: If we plan to run integration outside the CICD workflow, we could move this to the main .github/workflows

Comment (amorrow-flexion) [inline test/e2e/.github/workflows/playwright.yml:1]: We do not currently use this file within a workflow. Is this necessary to have?

File: test/e2e/playwright.config.ts (hunk 1 above)
Comment (amorrow-flexion) [reply test/e2e/playwright.config.ts:53]: remove, keep chrome

**Review by amorrow-flexion — APPROVED** [review]

**Review by fmaddenflx — APPROVED** [review]

**Pull request merged**